lib/galaxy/jobs/runners/kubernetes.py

```diff
@@ -95,7 +100,8 @@ def __init__(self, app, nworkers, **kwargs):
                 map=str, valid=lambda s: s == "$gid" or isinstance(s, int) or not s or s.isdigit(), default=None
             ),
             k8s_cleanup_job=dict(map=str, valid=lambda s: s in {"onsuccess", "always", "never"}, default="always"),
-            k8s_pod_retries=dict(map=int, valid=lambda x: int(x) >= 0, default=3),
+            k8s_pod_retries=dict(map=int, valid=lambda x: int(x) >= 0, default=1), # note that if the backOffLimit is lower, this paramer will have not effect.
```

Comment:
Should probably add a note to deprecate the pod retries on the job conf.

Reply:
Do we need both `k8s_pod_retries` and `k8s_job_spec_back_off_limit`? Can we combine the two into one so it's easier for users?

Reply:
I'm happy to remove k8s_pod_retries, it was a design I introduced in the first runner version, when I had no awareness of galaxy resubmissions. I was mostly leaving it here as a courtesy for deprecation for whomever might be using it, but we can remove.

Reply:
But I would rather remove altogether the first one than combining them. However, working out the logic for this to work well took me while and some experimentation, so if we can leave this removal for a further iteration so that I don't need to re-test that so much now, it would be great.